magento/inventory
# MC-17524: Child product is not addable to Cart from PLP if it does not have default source assigned

### Description (*)
Grouped Product has possibility to setup default qty of its units. It means if the number is setted such a Grouped product will be possible to add to Cart from Product Listing Page directly. In the Cart, accordingly, will appear all the grouped product items with qty as it was setted before. Though, if some of the units is out of stock, such units won't appear in product cart. This algorythm implements by OutOfStockFilter plugin.

But this plugin doesn't counts MSI extension. So that's why it is suggested to change the plugin with the new one which would be check Stock Status considering current website's stock in the MSI's implementation of GroupedProduct.

### Manual testing scenarios (*)
1. Create Grouped product with one Simple product (in stock, with positive qties)
2. Create Custom source.
3. Create Stock, link it with the Custom source and set a channel to Default Website. (therefore Default Website will work with the Custom stock and not with the Default Stock anymore)
4. In the before created Simple product set qty of default source as 0, and assing it to the Custom source with In stock status and positive qty.
5. Find the Grouped product in the Product Listing page and add it to cart using an appropriative button.

Expected result:
The Simple product will appears in the Shopping cart.

Actual result:
Success message will appears and no items will shown in the Shopping cart.

### Questions or comments
Pay attention to how i check the products is salable or not, maybe there is another to do it for whole set of items in one request to database.

### Contribution checklist (*)
